File: app/src/main/java/org/metabrainz/mobile/api/data/search/entity/ArtistCredit.java
package org.metabrainz.mobile.api.data.search.entity;

public class ArtistCredit {
    private String name;
    private String joinphrase;
    private Artist artist;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getJoinphrase() {
        return joinphrase;
    }

    public void setJoinphrase(String joinphrase) {
        this.joinphrase = joinphrase;
    }

    public Artist getArtist() {
        return artist;
    }

    public void setArtist(Artist artist) {
        this.artist = artist;
    }
}

File: app/src/main/java/org/metabrainz/android/data/sources/api/entities/ArtistCredit.kt
package org.metabrainz.android.data.sources.api.entities

import org.metabrainz.android.data.sources.api.entities.mbentity.Artist

class ArtistCredit {
    var name: String? = null
    var joinphrase: String? = null
    var artist: Artist? = null
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o !is ArtistCredit) return false
        val that = o
        return name == that.name && joinphrase == that.joinphrase && artist == that.artist
    }
}